lib/web_ui/lib/src/engine/html/recording_canvas.dart

```diff
@@ -515,6 +515,22 @@ class RecordingCanvas {
         drawRRect(rrect, paint);
         return;
       }
+      // Use drawRect for straight line paths painted with a zero strokeWidth
+      final ui.Rect? line = sPath.toStraightLine();
+      if (line != null && paint.strokeWidth == 0) {
+        final double left = math.min(line.left, line.right);
+        final double top = math.min(line.top, line.bottom);
+        final double width = line.width.abs();
+        final double height = line.height.abs();
+        if (line.width == 0 || line.height == 0) {
```

Comment:
Is this check necessary? This condition seems to be guaranteed by `toStraightLine`, see https://github.com/flutter/engine/blob/a76683e69f844904519043679d5f4874c469523f/lib/web_ui/lib/src/engine/html/path/path_ref.dart#L263

Reply:
Good catch! I updated the PR to remove this redundant check.